### PR TITLE
Add Smart amp feedback pipeline in nocodec topology

### PR DESCRIPTION
**Pull request by ranj063**
This is a replacement for #6799 and depends on the kernel PR https://github.com/thesofproject/linux/pull/4160

### CODE REVIEW

File: tools/topology/topology2/cavs-nocodec.conf
Comment (lgirdwood) [inline tools/topology/topology2/cavs-nocodec.conf:142]: I'm guessing this would be the dummy smart amp for CI tests ?

Comment (ranj063) [reply tools/topology/topology2/cavs-nocodec.conf:142]: right this is with the smart_amp_test module

File: tools/topology/topology2/include/common/pin_format.conf
Comment (ujfalusi) [inline tools/topology/topology2/include/common/pin_format.conf:17]: value of 0 as index is valid? As far as I know the pin0 format is derived from the generic base format, so index = 0 is ignored.

File: tools/topology/topology2/include/common/pin_format.conf
Comment (kv2019i) [inline tools/topology/topology2/include/common/pin_format.conf:150]: input/output?

File: tools/topology/topology2/include/common/tokens.conf
Comment (kv2019i) [inline tools/topology/topology2/include/common/tokens.conf:125]: Allowing for multiple types of init payload is something we seem to need, so this seems more practical approach. It would be nice to have a pointer where the enumerated options are documented. This PR now defines "1".

**Review by kv2019i — COMMENTED** [review]
Looks good. I'll wait until other folks working on this area are good with this. @aiChaoSONG  @ujfalusi @libinyang ....

File: tools/topology/topology2/include/common/tokens.conf
Comment (ranj063) [reply tools/topology/topology2/include/common/tokens.conf:125]: Yes, fixed now.

File: tools/topology/topology2/include/common/pin_format.conf
Comment (ranj063) [reply tools/topology/topology2/include/common/pin_format.conf:17]: thats right pin 0 will be ignored. Now changed the min to 1

**Comment by ranj063** [comment]
Same comment as the kernel PR. The audio formats in the current tplg's need to be reworked to make them extensible for multiple input/output pins. This PR needs to wait until that is done.

File: tools/topology/topology2/include/components/smart_amp.conf
Comment (ujfalusi) [inline tools/topology/topology2/include/components/smart_amp.conf:74]: What I would like to have is a line brake above this with a comment:
`				# smart amp payload`
Above the comment we have the ABI header, it is just easier to see

Comment (ranj063) [reply tools/topology/topology2/include/components/smart_amp.conf:74]: @ujfalusi I cannot add that comment here as the data is a string.

**Review by plbossart — APPROVED** [review]
I didn't see anything that would look wrong but my understanding of topology isn't exactly stellar.

**Comment by ranj063** [comment]
Please don't merge this before the rimage and kernel dependencies are merged

**Review by lgirdwood — APPROVED** [review]

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by mengdonglin** [comment]
> SOFCI TEST

@ranj063 It seems the kernel and rimage dependencies all got merged. So we only need to fix the CI failure, right?

@aiChaoSONG  Can you please check if we have observed the same error in local test?

**Comment by kv2019i** [comment]
This depends on https://github.com/thesofproject/sof/pull/7069

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by kv2019i** [comment]
Removed DNM prefix after #7099 and #7100 merged. Proceeding with merge.

**Pull request merged**
